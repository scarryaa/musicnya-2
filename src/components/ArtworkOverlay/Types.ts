import { Accessor, JSX } from 'solid-js'

export type ArtworkOverlayProps = {
  type: ArtworkOverlayType
  isLink: boolean
  link?: string
  children?: JSX.Element
  isVisible?: Accessor<boolean>
  playClick?: (e) => void
  moreClick?: (e) => void
  roundBottomCorners?: boolean
  rounded?: boolean
}

export enum ArtworkOverlayType {
  PLAY_AND_MORE,
  PLAY,
  MORE,
  NONE
}
