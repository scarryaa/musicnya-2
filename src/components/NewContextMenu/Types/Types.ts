export enum ContextMenuItem {
  GRID = 'grid',
  LIST = 'list',
  IMMERSIVE = 'immersive',
  MINI_PLAYER = 'miniPlayer',
  SHARE = 'share',
  PROPERTIES = 'properties',
  CREATE_STATION = 'createStation',
  GO_TO_ARTIST = 'goToArtist',
  GO_TO_ALBUM = 'goToAlbum',
  SHUFFLE = 'shuffle',
  REMOVE_FROM_QUEUE = 'removeFromQueue',
  ADD_TO_PLAYLIST = 'addToPlaylist',
  FAVORITE_QUICK = 'favoriteQuick',
  CREATE_STATION_QUICK = 'createStationQuick',
  LOVE_QUICK = 'loveQuick',
  DISLIKE_QUICK = 'dislikeQuick',
  ADD_TO_LIBRARY_QUICK = 'addToLibraryQuick',
  PLAY_NEXT_QUICK = 'playNextQuick',
  PLAY_LAST_QUICK = 'playLastQuick'
}
