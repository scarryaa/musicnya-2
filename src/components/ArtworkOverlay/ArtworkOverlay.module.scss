@import '../../styles/mixins/mixins';

.artwork-overlay {
  position: relative;
  width: 100%;
  height: 100%;

  &-link {
    @include flex-layout(row, space-between, flex-end);

    width: 100%;
    height: 100%;
    position: absolute;
    inset: 0;
    opacity: 0;
    background-color: $color-overlay-bg;
    border-radius: $app-border-radius;
    transition: opacity $transition-fast ease-in-out;
    padding: 0.6rem 0.5rem;
    cursor: default;

    &-flex-end {
      @include flex-layout(row, flex-end, flex-end);
    }

    &-play-button {
      display: flex;
      justify-content: center;
      align-items: center;
      background-color: $color-overlay-bg;
      border-radius: 50%;
      padding: 1rem;
      width: 2rem;
      height: 2rem;
      transition: background-color $transition-fast ease-in-out;

      &:hover {
        background-color: var(--color-primary);
      }
    }

    &-more-button {
      display: flex;
      justify-content: center;
      align-items: center;
      background-color: $color-overlay-bg;
      border-radius: 50%;
      padding: 1rem;
      width: 2rem;
      height: 2rem;
      transition: background-color $transition-fast ease-in-out;

      &:hover {
        background-color: var(--color-primary);
      }
    }
  }

  &:hover {
    .artwork-overlay-link {
      opacity: 1;
    }
  }
}
