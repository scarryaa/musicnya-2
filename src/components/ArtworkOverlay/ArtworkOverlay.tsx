import { BasicOverlay } from './Components/BasicOverlay'
import { MoreOverlay } from './Components/MoreOverlay'
import { PlayAndMoreOverlay } from './Components/PlayAndMoreOverlay'
import { PlayOverlay } from './Components/PlayOverlay'
import { ArtworkOverlayProps, ArtworkOverlayType } from './Types'

export const ArtworkOverlay = (props: ArtworkOverlayProps) => {
  switch (props.type) {
    case ArtworkOverlayType.PLAY_AND_MORE:
      return <PlayAndMoreOverlay {...props} />
    case ArtworkOverlayType.MORE:
      return <MoreOverlay {...props} />
    case ArtworkOverlayType.PLAY:
      return <PlayOverlay {...props} />
    case ArtworkOverlayType.NONE:
    default:
      return <BasicOverlay {...props} />
  }
}
