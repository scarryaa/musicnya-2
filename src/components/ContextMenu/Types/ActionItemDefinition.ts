import { IconDefinition } from '@fortawesome/free-solid-svg-icons'

export type ActionItemDefinition = {
  item: {
    icon: IconDefinition
    action: () => void
    label: string
    hasSubMenu?: boolean
    onMouseEnter?: (e: MouseEvent, id: string) => void
    onMouseLeave?: () => void
    id?: string
  }
}
