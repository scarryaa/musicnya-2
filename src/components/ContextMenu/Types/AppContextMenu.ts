import { generateMenuItems } from '../Helpers/ContextMenuHelpers'
import { ContextMenuItem } from '../Types'

export const appContextMenu = () => {
  return generateMenuItems(
    null,
    'app',
    [ContextMenuItem.IMMERSIVE, ContextMenuItem.MINI_PLAYER],
    []
  )
}
