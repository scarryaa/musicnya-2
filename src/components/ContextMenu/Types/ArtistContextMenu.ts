import { generateMenuItems } from '../Helpers/ContextMenuHelpers'
import { ContextMenuItem } from '../Types'

export const artistContextMenu = (
  id: string,
  disabled: boolean,
  isFavoritedState: boolean
) => {
  return generateMenuItems(
    id,
    'app',
    [
      ContextMenuItem.CREATE_STATION_QUICK,
      ContextMenuItem.FAVORITE_QUICK,
      ContextMenuItem.SHARE,
      ContextMenuItem.PROPERTIES
    ],
    [disabled, isFavoritedState]
  )
}
