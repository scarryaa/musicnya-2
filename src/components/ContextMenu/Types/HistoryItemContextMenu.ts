import { generateMenuItems } from '../Helpers/ContextMenuHelpers'
import { ContextMenuItem } from '../Types'

export const historyItemContextMenu = (
  id: string,
  subtype: string,
  disabled: boolean,
  isLovedState = false,
  isDislikedState = false,
  inLibraryState = false
) => {
  const additionalParams = [disabled, isLovedState, isDislikedState, inLibraryState]
  switch (subtype) {
    case 'stations':
      return generateMenuItems(
        id,
        subtype,
        [
          ContextMenuItem.LOVE_QUICK,
          ContextMenuItem.DISLIKE_QUICK,
          ContextMenuItem.SHARE,
          ContextMenuItem.PROPERTIES
        ],
        additionalParams
      )
    case 'playlists':
      return generateMenuItems(
        id,
        subtype,
        [
          ContextMenuItem.ADD_TO_LIBRARY_QUICK,
          ContextMenuItem.LOVE_QUICK,
          ContextMenuItem.DISLIKE_QUICK,
          ContextMenuItem.PLAY_NEXT_QUICK,
          ContextMenuItem.PLAY_LAST_QUICK,
          ContextMenuItem.ADD_TO_PLAYLIST,
          ContextMenuItem.SHUFFLE,
          ContextMenuItem.SHARE,
          ContextMenuItem.PROPERTIES
        ],
        additionalParams
      )
    default:
      return generateMenuItems(
        id,
        subtype,
        [
          ContextMenuItem.ADD_TO_LIBRARY_QUICK,
          ContextMenuItem.LOVE_QUICK,
          ContextMenuItem.DISLIKE_QUICK,
          ContextMenuItem.PLAY_NEXT_QUICK,
          ContextMenuItem.PLAY_LAST_QUICK,
          ContextMenuItem.ADD_TO_PLAYLIST,
          ContextMenuItem.CREATE_STATION,
          ContextMenuItem.GO_TO_ARTIST,
          ContextMenuItem.GO_TO_ALBUM,
          ContextMenuItem.SHARE,
          ContextMenuItem.PROPERTIES
        ],
        additionalParams
      )
  }
}
