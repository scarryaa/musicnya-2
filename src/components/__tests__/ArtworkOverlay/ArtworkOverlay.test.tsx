import { Router, useLocation } from '@solidjs/router'
import { createEffect, createSignal } from 'solid-js'
import { fireEvent, render, screen } from '@solidjs/testing-library'
import '@testing-library/jest-dom'
import { ArtworkOverlay } from '../../ArtworkOverlay/ArtworkOverlay'
import { ArtworkOverlayType } from '../../ArtworkOverlay/Types'

describe('ArtworkOverlay', () => {
  it('renders with PLAY_AND_MORE type', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('play-button')).toBeInTheDocument()
    expect(screen.queryByTestId('more-button')).toBeInTheDocument()
  })

  it('renders with MORE type', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('play-button')).not.toBeInTheDocument()
    expect(screen.queryByTestId('more-button')).toBeInTheDocument()
  })

  it('renders with NONE type', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.NONE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('play-button')).not.toBeInTheDocument()
    expect(screen.queryByTestId('more-button')).not.toBeInTheDocument()
  })

  it('renders with link', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={true}
          link="/media/albums/1"
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).toBeInTheDocument()
  })

  it('renders with children', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        >
          <div data-testid="children" />
        </ArtworkOverlay>
      </Router>
    ))

    expect(screen.queryByTestId('children')).toBeInTheDocument()
  })

  it('renders with roundBottomCorners', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          roundBottomCorners={true}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).toHaveClass(
      'artwork-overlay-link--round-bottom-corners'
    )
  })

  it('renders without roundBottomCorners', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          roundBottomCorners={false}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).not.toHaveClass(
      'artwork-overlay-link--round-bottom-corners'
    )
  })

  it('renders with rounded set to true', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          rounded={true}
          roundBottomCorners={false}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).toHaveStyle({
      'border-radius': '50%'
    })
  })

  it('renders with rounded set to false', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          rounded={false}
          roundBottomCorners={false}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).not.toHaveStyle({
      'border-radius': '50%'
    })
  })

  it('renders with isVisible set to false', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => false}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).toHaveStyle({
      opacity: 0
    })
  })

  it('renders with isVisible set to true', () => {
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(screen.queryByTestId('artwork-overlay-link')).toHaveStyle({
      opacity: 1
    })
  })

  it('calls playClick when play button is clicked', () => {
    const playClick = jest.fn()
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={playClick}
          moreClick={() => {}}
        />
      </Router>
    ))

    screen.getByTestId('play-button').click()
    expect(playClick).toHaveBeenCalledTimes(1)
  })

  it('calls moreClick when more button is clicked', () => {
    const moreClick = jest.fn()
    render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={moreClick}
        />
      </Router>
    ))

    screen.getByTestId('more-button').click()
    expect(moreClick).toHaveBeenCalledTimes(1)
  })

  const TestComponent = () => {
    const location = useLocation()
    const [currentPath, setCurrentPath] = createSignal(location.pathname)

    // Update path on route change
    createEffect(() => {
      setCurrentPath(location.pathname)
    })

    return <div>{currentPath()}</div> // Displays the current path
  }

  it('navigates to /media/albums/1 when link is clicked', async () => {
    render(() => (
      <Router>
        <TestComponent />
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={true}
          link="/media/albums/1"
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    // Click the link
    fireEvent.click(screen.getByTestId('artwork-overlay-link'))

    // Wait for the state to update and the router to navigate
    await new Promise(r => setTimeout(r, 0))

    // Check if the path has changed
    expect(screen.getByText('/media/albums/1')).toBeInTheDocument()
  })

  it('does not navigate when link is clicked and isLink is false', async () => {
    render(() => (
      <Router>
        <TestComponent />
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link="/media/albums/1"
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    // Click the link
    fireEvent.click(screen.getByTestId('artwork-overlay-link'))

    // Wait for the state to update and the router to navigate
    await new Promise(r => setTimeout(r, 0))

    // Check if the path has changed
    expect(screen.getByText('/')).toBeInTheDocument()
  })
})

describe('ArtworkOverlay Snapshot', () => {
  it('matches snapshot with PLAY_AND_MORE type', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with MORE type', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with NONE type', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.NONE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with link', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={true}
          link="/media/albums/1"
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with children', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
        >
          <div data-testid="children" />
        </ArtworkOverlay>
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with roundBottomCorners', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          roundBottomCorners={true}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot without roundBottomCorners', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          roundBottomCorners={false}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with rounded set to true', () => {
    const { container } = render(() => (
      <Router>
        <ArtworkOverlay
          type={ArtworkOverlayType.PLAY_AND_MORE}
          isLink={false}
          link=""
          isVisible={() => true}
          playClick={() => {}}
          moreClick={() => {}}
          rounded={true}
          roundBottomCorners={false}
        />
      </Router>
    ))

    expect(container).toMatchSnapshot()
  })
})
