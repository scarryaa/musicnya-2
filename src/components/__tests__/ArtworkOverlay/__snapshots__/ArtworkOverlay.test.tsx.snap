// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtworkOverlay Snapshot matches snapshot with MORE type 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; justify-content: flex-end; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with NONE type 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 0;"
    />
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with PLAY_AND_MORE type 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with children 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    <div
      data-testid="children"
    />
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with link 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners inactive"
      data-testid="artwork-overlay-link"
      href="/media/albums/1"
      link=""
      style="pointer-events: auto; opacity: 1; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with roundBottomCorners 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link artwork-overlay-link--round-bottom-corners"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot with rounded set to true 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 50%;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;

exports[`ArtworkOverlay Snapshot matches snapshot without roundBottomCorners 1`] = `
<div>
  <div
    class="artwork-overlay"
  >
    <a
      aria-current="page"
      class="artwork-overlay-link"
      data-testid="artwork-overlay-link"
      href="/#"
      link=""
      style="pointer-events: none; opacity: 1; border-radius: 0;"
    >
      <button
        class="artwork-overlay-link-play-button"
        data-testid="play-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(192 256)"
            transform-origin="96 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
                fill="var(--color-white)"
                transform="translate(-192 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
      <button
        class="artwork-overlay-link-more-button"
        data-testid="more-button"
      >
        <svg
          aria-hidden="true"
          class="solid-fa"
          style="height: 1em; font-size: 1em; vertical-align: -.125em; transform-origin: center; overflow: visible;"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(224 256)"
            transform-origin="112 0"
          >
            <g
              transform="translate(0,0) scale(1,1)"
            >
              <path
                d="M8 256a56 56 0 1 1 112 0A56 56 0 1 1 8 256zm160 0a56 56 0 1 1 112 0 56 56 0 1 1 -112 0zm216-56a56 56 0 1 1 0 112 56 56 0 1 1 0-112z"
                fill="var(--color-white)"
                transform="translate(-224 -256)"
              />
            </g>
          </g>
        </svg>
      </button>
    </a>
    
  </div>
</div>
`;
